### ISSUE TITLE
Chatlog Rows

### ISSUE THREAD

**Issue by wassie** [AoKMiKeY/Echelon#1]
Original request from FrikFry:
+1 for me

The chatlogs page and the clientdetails chatlogs do not have any ability to see beyond the initial amount displayed. IE, I believe they should have navigation buttons that allow mutliple pages of data to be accessed.

I'm not sure if this is a bug in the code or something that was intended by eire.

**Comment by AoKMiKeY** [comment]
I am pretty sure it is by design but I fully intend to remove this limitation. I will be adding up to a weeks worth of paged logs and then a menu to allow more logs to be shown.

Thanks for the suggestion

**Comment by wassie** [comment]
hi Mikey, did you had some time to look into Echelon?

**Comment by AoKMiKeY** [comment]
@wassie Yep. I am looking into it now. There is a lot of work to be done so do not expect instant changes but I have taken everything into account. I have added two milestones one for design and one for a strong and more stable base code. I am currently working on splitting echelon into two. An admin side and a less admin side. So the user management ( admins ) and permission. The set up and the install + other stuff will all be kept under a different part of the site. It should make it easier to use. 

Lots of plans. Lots to do :P You should see some new pushs soon. I will make releases when I think it is stable enough to be used on a production server. But for now this is still very alpha seeing as the current system does not work. :P

**Comment by wassie** [comment]
great Michael, I really appreciate that! I had some issue today within my gameserver and I had to spit all chatlog table to read what was going on.... keep the good work going!

**Comment by MichaelHillcox** [comment]
@wassie Care to add any more info to what happened there? 
seeing as my development on this project has gone up. I will be forking the old v2 version of echelon and pushing much smaller updates to that. Just to fix up existing issues.

**Comment by wassie** [comment]
@MichaelHillcox do you mean what happened in my gameserver? well, a lot of arguments between players and I wanted to be able to read the discussion between them. But I couldn't read back because only 100 lines are showing in chatlog.
So you are developping also?

**Comment by MichaelHillcox** [comment]
@wassie Oh yeah the 100 lines thing. its a bit stupid. Ideally I want people to move onto the new version I am making because I am taking a completely different approach to things. but I will go and fork a copy of the original v2 to see whats going on and what needs fixing but anything that is a feature I will only be adding to mine. Hopefully Mine will be a huge amount more efficient and clever about how it works. 

I will set up a webpage or something to use as a mini blog from the development of this application.

You can just look at the chatlogs in the database. I would assume you are using mysql? if so they you can just look at it if you are using something like phpmyadmin.

**Comment by wassie** [comment]
hi Michael,

how does the developing going? do you have something to share yet? I would be really happy!

**Comment by MichaelHillcox** [comment]
@wassie Development is going well. I have stopped pushing to github for a little bit just because I am only working on very annoying little boring bits. I will sync the commits soon. 

I have also decided to once again fork Echelon V2 and give it a tiny facelift but mainly focus on giving its code a stronger and more stable base to run on along with adding small improvements. I will not be bumping up the feature list or doing huge overhauls. Mainly because I want this to be the successor of Echelon V2. 

I will be reaching out to the devs at Big Brother Bot to see if I can get my own version officially supported once I have more of a base to work with. At the same time as all of this I am also looking at moving away from php to Node.js or Ruby. My main concern with php is that its not made for this sort of application.

**Comment by wassie** [comment]
Sounds good @MichaelHillcox ! Couldn't tell if this project still got your attention. Nice to hear that you put so much effort in it, well done!

About your remark about moving away from PHP to Node.js or Ruby, is it possible to run applications at a webhost while made in Node.js or Ruby?

When can I try a new release?

**Comment by MichaelHillcox** [comment]
@wassie I will make a pre-alpha release soon. I will also add a note of how to port your current system to mine. There are database changes. I am creating a update tool on the site to do all the heavy work for you though. I would not suggest using this for a production server.

I guess moving to Node.js would impact all the people not running there site on a dedicated server. Some webhosts have support for this but most wont. Almost all support Ruby so that is likely the best option to me so far to keep up a good line of support for everyone.
